include/networkit/graph/GraphTools.hpp

```diff
@@ -129,13 +129,79 @@ std::pair<node, node> size(const Graph &G) noexcept;
 double density(const Graph &G) noexcept;
 
 /**
- * Returns the volume (sum of the out-degree of all nodes) of the graph.
+ * Returns the volume of the input graph.
  *
  * @param G The input graph.
  * @return Volume of the graph.
  */
 double volume(const Graph &G);
 
+/**
+ * Returns the volume (sum of the out-degree of all nodes) of a range
+ * of nodes.
+ *
+ * @param G The input graph.
+ * @param first,last The range of nodes to insert
+ * @return Volume of the graph.
+ */
+template <class InputIt>
+double volume(const Graph &G, InputIt first, InputIt last) {
+    double outVolume = 0.0;
+    bool countSelfLoopsTwice = !G.isDirected();
+    count nodesInGraph = 0;
+
+    // Check for nodes in range, which are part of the graph
+    for (auto actIt = first; actIt != last; ++actIt) {
+        if (G.hasNode(*actIt))
+            nodesInGraph++;
+    } 
+
+    // Compute volume of subgraph if the subset is really a subset
+    if (nodesInGraph != G.numberOfNodes()) {
+        for (auto actIt = first; actIt != last; ++actIt) {
+            outVolume += G.weightedDegree(*actIt, countSelfLoopsTwice);
+        }
+    // Otherwise compute volume of the complete graph
+    } else {
+        outVolume = volume(G);
+    }
```

Comment:
Does it really make sense to distinguish these two cases? I think that you can simply compute the volume as
```
for (auto actIt = first; actIt != last; ++actIt) {
    outVolume += G.weightedDegree(*actIt, countSelfLoopsTwice);
}
```
and we can assume that the input range is already a set, we should not check that.